src/main/scala/dotty/dokka/InheritanceInformationTransformer.scala

```diff
@@ -0,0 +1,89 @@
+package dotty.dokka
+
+import org.jetbrains.dokka.transformers.documentation.DocumentableTransformer
+import org.jetbrains.dokka.model._
+import collection.JavaConverters
+import collection.JavaConverters._
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.properties._
+
+class InheritanceInformationTransformer(val ctx: DokkaContext) extends DocumentableTransformer{
+    override def invoke(original: DModule, context: DokkaContext): DModule = {
+        val supertypes = getSupertypes(original)
+        val subtypes = getSubtypesMap(supertypes)
+        completeInheritanceInformation(original, subtypes)
+    }
+
+    private def getSupertypes(d: Documentable): List[(DRI, DRI)] = d match {
+        case m: DModule => m.getPackages.asScala.toList.flatMap(p => getSupertypes(p))
+        case p: DPackage => p.getClasslikes.asScala.toList.flatMap(c => getSupertypes(c))
+        case c: DClass => c.get(InheritanceInfo).parents.map(p => (c.getDri, getTypeDRI(p))) ++ c.getClasslikes.asScala.toList.flatMap(c => getSupertypes(c))
+        case other => List.empty
+    }
+
+    private def getTypeDRI(b: Bound) = b match {
+        case t: TypeConstructor => t.getDri
+        case other => throw IllegalStateException(s"Supertype without DRI: $b")
+    }
+
+    private def getSubtypesMap(supertypesList: List[(DRI, DRI)]): Map[DRI, List[DRI]] = supertypesList
+        .map( (a,b) => (b,a) )
+        .groupBy( (a,b) => a )
+        .map{
+            case (key, l) => (key, l.map(_(1)))
+        }.toMap
+
+    private def completeInheritanceInformation[T <: Documentable](d: T, subtypes: Map[DRI, List[DRI]]): T = d match {
+        case m: DModule => m.copy(
+            m.getName,
+            m.getPackages.asScala.map(m => completeInheritanceInformation(m,subtypes)).asJava,
+            m.getDocumentation,
+            m.getExpectPresentInSet,
+            m.getSourceSets,
+            m.getExtra
+        ).asInstanceOf[T]
```

Comment:
That's probably bug of Dotty as it fails to compile without casting because of type incompatibility: Required: T, found: DModule

Reply:
Discussed in private: this is actually because `DModule` etc. are open classes. We don't have a guarantee that `m.copy` returns an instance of `m`'s class, and we don't have a guarantee in this branch that `T === DModule`. This code is morally fine however, so I wouldn't worry too much about it.